### PR TITLE
R 1.0.0

### LINKED ISSUE #145: make package CRAN-ready
- [x] Get rid of notes.
- [x] Consider changing api tests to not format in temp dir since it creates problems with appveyor.
- [x] Make tests passing quickly enough?
- [x] Wait for krlmlr/utf8 to be released on CRAN and remove `Remotes:` in DESCRIPTION
- [x] Clarify LICENCE with `usethis()`? ([Link](http://r-pkgs.had.co.nz/release.html))
- [x] Completing cran-release-notes.md
- [x] Update NEWS.md (describe every exported function in order of relevance, see pillar).
- [ ] Create a pre-release.
- [x] Bump version to 0.1.0
- [x] `devtools::check_spelling()`
- [x] Delete all man files for unexported functions.
- [x] Make sure every man page of exported functions has an example.

### PR DESCRIPTION
**Pull request by lorenzwalthert**
Not to be merged. Aims to close #145.

### CODE REVIEW

**Comment by lorenzwalthert** [comment]
This is the result from `devtools::build_win()`. Just this one note 
> checking CRAN incoming feasibility ... NOTE

https://win-builder.r-project.org/0uoMp9FDVJVz/00check.log

All the other things from #145 are incorporated, so I think we are ready for the submission.

**Comment by lorenzwalthert** [comment]
Also, @krlmlr can you udpate the pkgdown page?

**Comment by lorenzwalthert** [comment]
Also, @krlmlr can you update the pkgdown page before the submission?

**Comment by lorenzwalthert** [comment]
For `NEWS.md`: Do we throw away all previous entries in the file so only the API description is left?

**Comment by krlmlr** [comment]
Yes, we can remove old news.

Added deploy key and restarted Travis job https://travis-ci.org/r-lib/styler/jobs/311941953, which should update pkgdown.

**Comment by krlmlr** [comment]
Restarted job https://travis-ci.org/r-lib/styler/jobs/311951016 instead, this is the correct 0.0-11 release.

**Comment by krlmlr** [comment]
Built pkgdown manually, for some reason the rebuild on Travis CI doesn't catch all changes.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: styler
-Title: Non-invasive Pretty Printing of R code
-Version: 0.0-11
+Title: Non-Invasive Pretty Printing of R code
```
[the remark this answers is not in the document]
Comment (lorenzwalthert) [reply DESCRIPTION]: Right...

File: NEWS.md [hunk 2]
```diff
@@ -1,187 +1,74 @@
...
-
-
-## styler 0.0-1 (2017-06-15)
-
-Initial release, work in progress.
-
-- Create and serialize nested parse data.
-- Internal support for indention of expressions with parentheses.
-- Adding and removing spaces around operators on flat parse data.
+## styler 0.1.0 (2017-12-05)
```
[the remark this answers is not in the document]
Comment (lorenzwalthert) [reply NEWS.md]: Sure.

File: NEWS.md [hunk 3]
```diff
@@ -1,187 +1,74 @@
...
+
+
+style_file(path, 
+  ..., style = tidyverse_style, transformers = style(...)
+)
+
+style_text(text, ..., style = tidyverse_style, transformers = style(...))
+```
+
+### style guide creators
```
[the remark this answers is not in the document]
Comment (lorenzwalthert) [reply NEWS.md]: Ok, moved them up.

File: NEWS.md [hunk 4]
```diff
@@ -1,187 +1,74 @@
...
+specify_reindention(
+  regex_pattern = NULL, 
+  indention = 0, 
+  comments_only = TRUE
+)
+```
+
+These are helper functions that simply forward to the helper functions above.
+```
+tidyverse_reindention()
```
[the remark this answers is not in the document]
Comment (lorenzwalthert) [reply NEWS.md]: Ok.

File: NEWS.md [hunk 5]
```diff
@@ -1,187 +1,74 @@
...
+)
+
+style_text(text, ..., style = tidyverse_style, transformers = style(...))
+```
+
+### style guide creators
+This function is used to create a style guide.
+```
+create_style_guide(
+  initialize = initialize_attributes,
```
[the remark this answers is not in the document]
Comment (lorenzwalthert) [reply NEWS.md]: Yes, thanks, forgot about that. I think we may rename it to `default_style_guide_attributes()` then and export it.

**Comment by lorenzwalthert** [comment]
Can't figure out why the appveyor buid fails now. Maybe currently updating to 3.4.3?

File: NEWS.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lorenzwalthert) [reply NEWS.md]: Yes, thanks, forgot about that. I think we may rename it to default_style_guide_attributes() then and export it.

**Comment by lorenzwalthert** [comment]
New winbuild results
https://win-builder.r-project.org/81q44YxOBi0k/

> checking CRAN incoming feasibility ... NOTE

Code is now capitalised in the DESCRIPTION, so everything should be fine now.

File: NEWS.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lorenzwalthert) [reply NEWS.md]: yes -.-

**Comment by lorenzwalthert** [comment]
I also rephrased the help file for `default_style_guide_attributes()`.

**Comment by lorenzwalthert** [comment]
Are we set for the release?

**Comment by lorenzwalthert** [comment]
Should we also style the source code with styler prior to the CRAN submission?

File: NEWS.md [hunk 6]
```diff
@@ -1,187 +1,71 @@
...
+)
+tidyverse_reindention()
+tidyverse_math_token_spacing())
+```
+
+### style guide creators
+This function is used to create a style guide.
+```
+create_style_guide(
+  initialize = default_style_guide_attributes,
```
Comment (krlmlr) [inline NEWS.md]: Do we need parens here?

Comment (lorenzwalthert) [reply NEWS.md]: No there must not be braces, see the example in the docs
```r
set_line_break_before_curly_opening <- function(pd_flat) {
  op <- pd_flat$token %in% "'{'"
  pd_flat$lag_newlines[op] <- 1L
  pd_flat
}
set_line_break_before_curly_opening_style <- function() {
  create_style_guide(line_break = set_line_break_before_curly_opening)
}
style_text("a <- function(x) { ! x }", style = set_line_break_before_curly_opening_style)

```
I hope the following is clear enough:
```r
#' @param line_break A list of transformer functions that manipulate line_break
#'   information.
```
The other arguments are described in a similar fashion.

Do you think it's clear enough or should we describe each of these arguments more explicitly, i.e. like this:
```r
#' @param line_break A list of bare transformer function names that manipulate
#'   line_break information.
```

Comment (lorenzwalthert) [reply NEWS.md]: ?

Comment (krlmlr) [reply NEWS.md]: Oh, I see. Makes sense.

**Comment by krlmlr** [comment]
I don't mind postponing styling styler until after the CRAN release, but we can do it if you like.

**Pull request merged**